src/main/java/de/doubleslash/keeptime/view/ReportController.java

```diff
@@ -113,120 +115,118 @@
-         final Button bProjectReport = createProjectReport();
-         this.gridPane.add(bProjectReport, 1, rowIndex);
-
-         rowIndex++;
-
-         final ProjectReport pr = new ProjectReport(onlyCurrentProjectWork.size());
-         for (int j = 0; j < onlyCurrentProjectWork.size(); j++) {
-            final Work work = onlyCurrentProjectWork.get(j);
-            final String workedHours = DateFormatter
-                  .secondsToHHMMSS(DateFormatter.getSecondsBewtween(work.getStartTime(), work.getEndTime()));
-
-            final String currentWorkNote = work.getNotes();
-            pr.appendToWorkNotes(currentWorkNote);
-            final Label commentLabel = new Label(currentWorkNote);
-            commentLabel.setFont(FontProvider.getDefaultFont());
-            commentLabel.setWrapText(true);
-            this.gridPane.add(commentLabel, 0, rowIndex);
-
-            final Label fromTillLabel = new Label(DateFormatter.toTimeString(work.getStartTime()) + " - "
-                  + DateFormatter.toTimeString(work.getEndTime()));
-            fromTillLabel.setFont(FontProvider.getDefaultFont());
-            fromTillLabel.setWrapText(true);
-            this.gridPane.add(fromTillLabel, 1, rowIndex);
-
-            final Label workedHoursLabel = new Label(workedHours);
-            workedHoursLabel.setFont(FontProvider.getDefaultFont());
-            this.gridPane.add(workedHoursLabel, 2, rowIndex);
+         final HBox projectButtonBox = new HBox();
+         projectButtonBox.getChildren().add(createProjectReportButton(onlyCurrentProjectWork));
+         // TODO center Dot
 
-            final HBox clickDummy = new HBox();
-            final ContextMenu contextMenu = new ContextMenu();
-            final MenuItem editMenuItem = new MenuItem("edit");
+         final Circle circle = new Circle(9, project.getColor());
 
-            editMenuItem.setOnAction(e -> {
-               LOG.info(EDIT_WORK_DIALOG_TITLE);
-               final Dialog<Work> dialog = setupEditWorkDialog(work);
+         final TreeItem<TableRow> projectRow = new TreeItem<>(
+               new ProjectTableRow(project, projectWorkSeconds, projectButtonBox), circle);
 
-               final Optional<Work> result = dialog.showAndWait();
-
-               result.ifPresent(editedWork -> {
-                  controller.editWork(work, editedWork);
-
-                  this.update();
-               });
-            });
-
-            contextMenu.getItems().add(editMenuItem);
-
-            clickDummy.setOnContextMenuRequested(
-                  event -> contextMenu.show(clickDummy, event.getScreenX(), event.getScreenY()));
+         for (final Work w : onlyCurrentProjectWork) {
+            final HBox workButtonBox = new HBox();
+            workButtonBox.getChildren().add(createEditWorkButton(w));
+            final TreeItem<TableRow> workRow = new TreeItem<>(new WorkTableRow(w, workButtonBox));
+            projectRow.getChildren().add(workRow);
```

Comment:
the rows are now to high. can only fit < half rows than before. please decrease the height of the rows ( i guess the button is the problem)